src/blenderbim/blenderbim/bim/module/classification/data.py

```diff
@@ -0,0 +1,106 @@
+# BlenderBIM Add-on - OpenBIM Blender Add-on
+# Copyright (C) 2021 Dion Moult <dion@thinkmoult.com>
+#
+# This file is part of BlenderBIM Add-on.
+#
+# BlenderBIM Add-on is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# BlenderBIM Add-on is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with BlenderBIM Add-on.  If not, see <http://www.gnu.org/licenses/>.
+
+import bpy
+import blenderbim.tool as tool
+
+import ifcopenshell.util.date
+
+def refresh():
+    ClassificationsData.is_loaded = False
+
+class ClassificationsData:
+    is_loaded = False
+    data = {
+        "classifications" : {},
+        "library_file" : None,
+        "library_classifications" : {},
+    }
+
+    @classmethod
+    def load(cls):
+        cls._file = tool.Ifc.get()
+        cls.data["classifications"] = cls.get_classifications()  #list of available classifications, es. Omniclass, Uniclass,etc..
+        cls.is_loaded = True
+
+    @classmethod
+    def get_classifications(cls):
+        results = {}
+        for classification in cls._file.by_type("IfcClassification"):
+            info = classification.get_info()
+            if cls._file.schema == "IFC2X3" and info["EditionDate"]:
+                info["EditionDate"] = ifcopenshell.util.date(info.EditionDate).isoformat()
+            results = {
+                classification.id() : info,
+            }
+
+        return results
+
+    @classmethod
+    def get_referenced_source(cls, reference):
+        if reference.is_a("IfcClassification"):
+            return reference
+        elif reference.is_a("IfcClassificationReference") and reference.ReferencedSource:
+            return cls.get_referenced_source(reference.ReferencedSource)
+
+    @classmethod
+    def load_library(cls, filepath):
+        cls.data["library_file"] = ifcopenshell.open(filepath)
+        cls.data["library_classifications"] = {}
+        for classification in cls.data["library_file"].by_type("IfcClassification"):
+            cls.data["library_classifications"][classification.id()] = classification.Name
+
+
+class ClassificationReferencesData:
+    is_loaded = False
+    data = {
+        "products" : {},
+        "references" : {},
+        "library_references" : {},  #never called
+    }
+
+    @classmethod
+    def load(cls,  product_id=None):
```

Comment:
Signature must only be def load(cls), no other arguments allowed. This is because we cannot rely on any intelligence on the code calling the UI to be reloaded. The product is always the active object.